charts/signoz/values.yaml

```diff
@@ -2090,27 +2107,47 @@ k8s-infra:
           id: extract_metadata_from_filepath
           regex: '^.*\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[a-f0-9\-]+)\/(?P<container_name>[^\._]+)\/(?P<restart_count>\d+)\.log$'
           parse_from: attributes["log.file.path"]
+          output: add_cluster_name
+        # Add cluster name attribute from environment variable
+        - id: add_cluster_name
+          type: add
+          field: resource["k8s.cluster.name"]
+          value: EXPR(env("K8S_CLUSTER_NAME"))
+          output: move_stream
         # Rename attributes
         - type: move
+          id: move_stream
           from: attributes.stream
           to: attributes["log.iostream"]
+          output: move_container_name
         - type: move
+          id: move_container_name
           from: attributes.container_name
           to: resource["k8s.container.name"]
+          output: move_namespace
         - type: move
+          id: move_namespace
           from: attributes.namespace
           to: resource["k8s.namespace.name"]
+          output: move_pod_name
         - type: move
+          id: move_pod_name
```

Comment:
Why is this chaining needed? It makes sense at the start of the parsing to decide docker/crio/containerd but I don't see why everything has to be chained. It goes in the order it is defined already. If someone adds a new block, they must remember that it is chained, which is easy to miss.

Reply:
@srikanthccv agreed. It took us quiet some time ourselves to figure out this was required.

@nityanandagohain can you please look into this?

Reply:
@srikanthccv , so basically when someone wants to add a new block like they want to parse json logs after k8s attributes are parsed, then they will have to do the chaining for all the above operators we added. So chaining by default will help them to just focus on adding new operators instead of worrying about the ones we have added.

Also if they add new operators to the end, even if they don't chain them it will be run sequentially

Reply:
>add a new block like they want to parse json logs after k8s attributes are parsed, then they will have to do the chaining for all the above operators we added

Can you give some usecase for this? Why do they want to add the json parser just after k8s attributes?  

>chaining by default will help them to just focus on adding new operators instead of worrying about the ones we have added

This forces them to remember the detail that they are chained. If they forget about it and add an operator, they will have a hard time figuring it out. Prashant is one such persona.

Reply:
> This forces them to remember the detail that they are chained. If they forget about it and add an operator, they will have a hard time figuring it out. Prashant is one such persona.

Some of our operators were chained while some weren't because of which it was causing more confusion, if you see the entire config the routing part is chained while the remaining part was unchained because of which it was not consistent. I think this makes it more clear if the user wants to do something.

Reply:
> Can you give some usecase for this? Why do they want to add the json parser just after k8s attributes?

An example is adding nginx parser-> it will require chaining, but again for keeping it consistent we have added chaining.

Reply:
It makes sense why the routing part is chained because it needs to decide the format, and the rest is just sequential. A chaining that is the same as the sequential order is better not chained. I believe @prashant-shahi had to sync with you to understand why an operator was not getting added to logs. I am taking that as a reference to highlight for someone who just adds an operator (type, from, to) where they don't bother any order makes a difference.